src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,214 @@
+        if (bucket == null) {
+            return null;
+        }
+
+        if (Objects.equals(bucket.key, key)) {
+            return bucket.value;
+        } else {
+            while (bucket != null) {
+                if (Objects.equals(bucket.key, key)) {
+                    return bucket.value;
+                }
+                bucket = bucket.next;
+            }
+            return null;
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    public boolean containsKey(K key) {
+        for (Node<K, V> bucket : buckets) {
+            while (bucket != null) {
+                if (bucket.key.equals(key)) {
+                    return true;
+                }
+                bucket = bucket.next;
+            }
+        }
+        return false;
+    }
+
+    public boolean containsValue(V value) {
+        for (Node<K, V> bucket : buckets) {
+            while (bucket != null) {
+                if (bucket.value.equals(value)) {
+                    return true;
+                }
+                bucket = bucket.next;
+            }
+        }
+        return false;
+    }
+
+    public void clear() {
+        capacity = DEFAULT_CAPACITY;
+        buckets = new Node[DEFAULT_CAPACITY];
+        threshold = capacity * LOAD_FACTOR;
+    }
+
+    public V remove(K key) {
+        int index = key != null ? Math.abs(key.hashCode() % capacity) : 0;
+        Node<K, V> bucket = buckets[index];
+        V value = getValue(key);
+        if (Objects.equals(buckets[index].key, key)) {
+            buckets[index] = buckets[index].next;
+            return value;
+        }
+        while (bucket != null) {
+            if (Objects.equals(bucket.next.key, key)) {
+                bucket.next = bucket.next.next;
+                return value;
+            } else {
+                bucket = bucket.next;
+            }
+        }
+        return null;
+    }
+
+    public Set<K> keySet() {
+        Set<K> keySet = new HashSet<>();
+        for (Node<K, V> bucket : buckets) {
+            while (bucket != null) {
+                keySet.add(bucket.key);
+                bucket = bucket.next;
+            }
+        }
+        return keySet;
+    }
+
+    public Collection<V> values() {
+        Collection<V> values = new HashSet<>();
+        for (Node<K, V> bucket : buckets) {
+            while (bucket != null) {
+                values.add(bucket.value);
+                bucket = bucket.next;
+            }
+        }
+        return values;
+    }
+
+    public Set<Map.Entry<K,V>> entrySet() {
+        Set<Map.Entry<K, V>> entrySet = new HashSet<>();
+        for (Node<K, V> bucket : buckets) {
+            while (bucket != null) {
+                entrySet.add(new AbstractMap.SimpleEntry<>(bucket.key, bucket.value));;
+                bucket = bucket.next;
+            }
+        }
+        return entrySet;
+    }
+
+    private void resize() {
```

Comment:
this method should be simplified. Consider reusing your existing methods (put, for example)